### PR TITLE
feat: cache fee rate. delete calculateFee(amounts)

### PR DESCRIPTION
**Pull request by MussinaLin**
Delete calculateFee(amounts) function.
Gas estimate:
- calculateFee(amount): save 152 gas / call.
- calculateAmountWithFee(amounts):  save 165 gas / call with 3 elements in amounts.

### CODE REVIEW

File: src/fees/FeeCalculatorBase.sol [hunk 1]
```diff
@@ -28,20 +28,8 @@ abstract contract FeeCalculatorBase {
...
-    function calculateFee(uint256[] memory amounts) public view returns (uint256[] memory) {
-        for (uint256 i; i < amounts.length; ) {
-            amounts[i] = (amounts[i] * feeRate) / (_BPS_BASE + feeRate);
-            unchecked {
-                ++i;
-            }
-        }
-        return amounts;
+        uint256 feeRate_ = feeRate;
+        return (amount * feeRate_) / (_BPS_BASE + feeRate_);
```
Comment (zodahu) [inline src/fees/FeeCalculatorBase.sol:32]:
```suggestion
        uint256 _feeRate = feeRate;
        return (amount * _feeRate) / (_BPS_BASE + _feeRate);
```
As internal coding style said `Local variables has prefix _ if conflicts with storage variable`

Comment (MussinaLin) [reply src/fees/FeeCalculatorBase.sol]: Oh you are right. I confused with function parameters.

**Comment by zodahu** [comment]
LGTM

**Comment by JavanC** [comment]
LGTM

**Pull request merged**